active/0000-engines.md

```diff
@@ -32,117 +29,298 @@ Unfortunately, this approach leads to frequent conflicts around shared
 resources, and feedback from tests gets slower and slower as test suites
 grow in size.
 
-Engines allow each section of an application to be developed in
-isolation, including separate test suites and naming conventions. These
-isolated units can be composed together into a single app from the
-perspective of the user.
+A more modular approach is to break off elements of a single application into
+separate [addons](http://www.ember-cli.com/user-guide/#addons). Addons are
+essentially mixins for [ember-cli](http://www.ember-cli.com/) applications. In
+other words, the elements of an addon are merged with those of the application
+that includes them. While addons allow for distributed development, testing, and
+packaging, they do not provide the logical run-time separation required for
+developing completely independent "sections" of an application. Addons must
+function within the namespace, registry, and router of the application in which
+they are included.
 
-Additionally, engines allow addons to augment applications with
-off-the-shelf fully-functional UI. For example, imagine an
-authentication library that offers a prefab login screen that
-application developers can opt in to, dramatically increasing the speed
-at which prototypes can be developed.
+Engines provide an alternative to these approaches that allows for distributed
+development, testing, and packaging, _as well as_ logical run-time separation.
+Because engines are derived from applications, they can be just as
+full-featured. Each has its own namespace and registry. Even though engines are
+isolated from the applications that contain them, the boundaries between them
+allow for controlled sharing of resources.
 
-Or imagine a product like Discourse allowing you to mount their Ember
-app at `/forums/`, so the user needn't leave your app to read your
-discussion forum.
+Engines can be either "routable" or "route-less":
 
-# Detailed Design
+* Routable engines provide a routing map which can be integrated with the
+  routing maps of parent applications or engines. Routing maps are alway eager
+  loaded, which allows for deep linking into an engine's routes regardless of
+  whether the engine itself has been instantiated.
 
-Your app can gain access to an engine in one of two ways:
+* Route-less engines can isolate complex functionality that is not related to
+  routing (e.g. a chat engine in a sidebar). Route-less engines can be rendered
```

Comment:
I actually do mean route-less here. Route-less engines aren't mounted via a routing map so they must be mounted manually. See the section "Mounting route-less engines" (line 205) below.